contract/src/main.rs

```diff
@@ -499,90 +516,65 @@ pub extern "C" fn mint() {
-                NFTIdentifierMode::Hash => {
-                    let token_hashes: Vec<String> = owned_tokens
-                        .into_iter()
-                        .map(|identifier| identifier.get_hash().unwrap_or_revert())
-                        .collect();
-                    utils::upsert_dictionary_value_from_key(
-                        OWNED_TOKENS,
-                        &owned_tokens_item_key,
-                        token_hashes,
-                    )
-                }
+    // Update the page entry first
+    let all_owners_uref = utils::get_uref(
+        ALL_TOKEN_OWNERS,
+        NFTCoreError::MissingAllOwnersSeedURef,
+        NFTCoreError::InvalidAllOwnersSeedURef,
+    );
+
+    let page_record_width = utils::get_stored_value_with_user_errors::<u64>(
+        MAX_PAGE_NUMBER,
+        NFTCoreError::MissingMaximumNumberOfPages,
+        NFTCoreError::InvalidMaximumNumberOfPages,
+    );
+
+    let mut page_record =
+        match storage::dictionary_get::<Vec<bool>>(all_owners_uref, &owned_tokens_item_key)
+            .unwrap_or_revert()
+        {
+            Some(page_record) => page_record,
+            None => vec![false; page_record_width as usize],
+        };
+
+    let _ = core::mem::replace(&mut page_record[current_page_number as usize], true);
+
+    storage::dictionary_put(all_owners_uref, &owned_tokens_item_key, page_record);
+
+    // Update the individual page record.
+    let single_page_uref = utils::get_uref(
+        &format!("{}-{}", PAGE_DICTIONARY_PREFIX, current_page_number),
+        NFTCoreError::MissingPageUref,
+        NFTCoreError::InvalidPageUref,
+    );
+
+    let single_page_item_key =
+        utils::make_page_dictionary_item_key(&token_owner_key, current_page_number);
+
+    let mut single_page =
+        match storage::dictionary_get::<Vec<bool>>(single_page_uref, &single_page_item_key)
+            .unwrap_or_revert()
+        {
+            Some(single_page) => single_page,
+            None => {
+                vec![false; PAGE_SIZE as usize]
             }
-        }
-        None => {
-            match identifier_mode {
-                NFTIdentifierMode::Ordinal => utils::upsert_dictionary_value_from_key(
-                    OWNED_TOKENS,
-                    &owned_tokens_item_key,
-                    vec![token_identifier.get_index().unwrap_or_revert()],
-                ),
-                NFTIdentifierMode::Hash => utils::upsert_dictionary_value_from_key(
-                    OWNED_TOKENS,
-                    &owned_tokens_item_key,
-                    vec![token_identifier.clone().get_hash().unwrap_or_revert()],
-                ),
-            };
-        }
-    };
+        };
+
+    let _ = core::mem::replace(&mut single_page[current_page_index as usize], true);
+
+    storage::dictionary_put(single_page_uref, &single_page_item_key, single_page);
+
+    let receipt_address = Key::dictionary(single_page_uref, single_page_item_key.as_bytes());
```

Comment:
move this down to where you create the name & store it

Reply:
Moved down